dataflow/src/main/java/org/checkerframework/dataflow/expression/ValueLiteral.java

```diff
@@ -37,6 +37,58 @@ public ValueLiteral(TypeMirror type, Object value) {
         this.value = value;
     }
 
+    /**
+     * Returns the negation of this literal. Throws an exception if negation is not possible.
+     *
+     * @return the negation of this literal
+     */
+    public ValueLiteral negate() {
+        if (TypesUtils.isIntegralPrimitive(type)) {
+            if (value == null) {
+                throw new Error("null value of integral type " + type);
+            }
+            return new ValueLiteral(type, negateBoxedPrimitive(value));
+        }
+        throw new Error(String.format("cannot negate: %s type=%s", this, type));
+    }
+
+    /**
+     * Negate a boxed primitive.
+     *
+     * @param o a boxed primitive
+     * @return a boxed primitive that is the negation of the argument
+     */
+    private Object negateBoxedPrimitive(Object o) {
+        if (value instanceof Byte) {
+            return Byte.valueOf((byte) -((Byte) value).byteValue());
+        }
+        if (value instanceof Short) {
+            return Short.valueOf((short) -((Short) value).shortValue());
+        }
+        if (value instanceof Integer) {
+            return Integer.valueOf(-((Integer) value).intValue());
+        }
+        if (value instanceof Long) {
+            return Long.valueOf(-((Long) value).longValue());
```

Comment:
This doesn't work when parsing `-9223372036854775808L`, `Long.MIN_VALUE`.  I added a test case.